content/storage/bind-mounts.md

```diff
@@ -266,6 +266,39 @@ $ docker container stop devtest
 $ docker container rm devtest
 ```
 
+## Recursive mounts
+
+When you bind mount a path that itself contains mounts, those submounts are
+also included in the bind mount by default. This behavior is configurable,
+using the `bind-recursive` option for `--mount`.
+
+If the bind mount is read-only, the Docker Engine makes a best-effort attempt
+at making the submounts read-only as well. This is referred to as recursive
+read-only mounts. Recursive read-only mounts require Linux kernel version 5.12
+or later. If you're running an older kernel version, submounts are
+automatically mounted as read-write by default. Attempting to set submounts to
+be read-only on a kernel version earlier than 5.12, using the
+`bind-recursive=readonly` option, results in an error.
+
+Supported values for the `bind-recursive` option are:
+
+`bind-recursive=<enabled|true|1>` (default)
+: Enables recursive bind-mount.
+  Read-only mounts are made recursively read-only if kernel is v5.12 or later.
+  Otherwise, submounts are read-write.
+
+`bind-recursive=<disabled|false|0>`
+: Disables recursive bind-mounts.
+  Submounts are ignored (not included in the bind mount).
+
+`bind-recursive=writable`
+: Enables recursive bind-mounts.
+  Submounts are read-write.
+
+`bind-recursive=readonly`
+: Enables recursive bind-mounts.
+  Submounts are read-only.
```

Comment:
Perhaps we should mention that this produces an error if the kernel or doesn't support this feature (kernel versions older than v5.12).